services/bots/helpers.py

```diff
@@ -0,0 +1,32 @@
+from shared.config import get_config
+from shared.torngit.base import TokenType
+from shared.typings.oauth_token_types import Token
+
+from services.github import get_github_integration_token
+
+
+def get_dedicated_app_token_from_config(
+    service: str, token_type: TokenType
+) -> Token | None:
+    # GitHub can have 'dedicated_apps', and those are preferred
+    dedicated_app = get_config(service, "dedicated_apps", token_type.value, default={})
+    app_id = dedicated_app.get("id")
+    installation_id = dedicated_app.get("installation_id")
+    pem_exists = "pem" in dedicated_app
+    if app_id and installation_id and pem_exists:
+        actual_token = get_github_integration_token(
+            service,
+            app_id=app_id,
+            installation_id=installation_id,
+            pem_path=f"yaml+file://{service}.dedicated_apps.{token_type.value}",
+        )
+        return Token(key=actual_token, username=f"{token_type.value}_dedicated_app")
+
```

Comment:
I know it's implicit (by not having a return statement) that we return None if we don't go through the if statement, but I think it'll be more readable if we have an explicit return None here.